File: Data Structures & Algorithms/binary-search/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        return this._binarySearch(0, nums.length-1, nums, target);
    }

    _binarySearch(left, right, arr, target) {
        if(left > right) return -1;
        let mid = left + Math.floor((right-left)/2);

        if(arr[mid] === target) return mid;

        return target > arr[mid] ?
        this._binarySearch(mid+1, right, arr, target) : 
        this._binarySearch(left, mid-1, arr, target);

    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int i = 0;
        int j = numbers.length - 1;
        int sumOfTwo = 0;
        while( i < j ) {
            sumOfTwo = numbers[i] + numbers[j];
            if(sumOfTwo == target) {
                return new int[]{i+1,j+1};
            }
            if(sumOfTwo > target){
                j--;
            } else if(sumOfTwo < target){
                i++;
            }
        }
        return new int[]{};
    }
}
